crates/hulk_nao/src/camera.rs

```diff
@@ -82,9 +101,8 @@ impl Camera {
         bail!("too many unsuccessful waiting retries");
     }
 
-    fn reset(&mut self) -> Result<()> {
+    fn reset(&self) -> Result<()> {
         let _lock = self.i2c_head_mutex.lock();
-        self.camera.take();
```

Comment:
For the duration of the camera reset, the camera is not locked. What happens to other receivers requesting access to the camera? Should this `reset()` lock the camera for the duration of the reset?

Reply:
`reset()` is private